### PR TITLE
Prepare for release

### PR DESCRIPTION
**Pull request by mnazbro**
* Adds logging
* README update
* Removed unnecessary inheritance structure
* Added InstallPipPackages task for downstream consumers

### CODE REVIEW

**Comment by mnazbro** [comment]
@punya Can I get a CR?

File: README.md [hunk 1]
```diff
@@ -11,31 +11,49 @@ Apply the plugin to your project following
...
-
 Then invoke the `setupPython` task and use the resulting installation directory from `Exec` tasks:
 
 ```gradle
 task launchNotebook(type: Exec) {
     dependsOn 'setupPython'
-    executable "${miniconda.buildEnvironmentDirectory}/bin/ipython"
-    args 'notebook'
+    executable "${miniconda.buildEnvironmentDirectory}/bin/python"
+    args '-c', 'print("Hello world!)'
```
Comment (punya) [inline README.md]: Missing a closing double-quote.

File: README.md [hunk 2]
```diff
@@ -11,31 +11,49 @@ Apply the plugin to your project following
...
 
-You can customize where the Miniconda installer script is downloaded from by adding a dependency to the
-`minicondaInstaller` configuration. By default, it's downloaded from
-[`http://repo.continuum.io`](http://repo.continuum.io).
+| Name | Default Value | Description | Optional |
+| ---- | ------------- | ----------- | -------- |
+| minicondaVersion | N/A | The miniconda version which you want to use. See [the miniconda repo](https://repo.continuum.io/miniconda/) | false
+| packages | N/A | The conda packages you want installed into your conda environment. This list must contain at least one argument. | false
+| bootstrapDirectoryPrefix | `new File(System.getProperty('user.home'), '.miniconda')` | The root directory to put the root install of miniconda. This helps performance by caching the root environment by `pythonVersion` and `minicondaVersion`. | true
+| buildEnvironmentDirectory | new File(buildDir, 'miniconda') | The directory to place your specific miniconda environment. | true
```
Comment (punya) [inline README.md]: Line 38 uses a code font for the example, this one uses a normal font. We should be consistent.

File: src/main/java/com/palantir/python/miniconda/tasks/BootstrapPython.java [hunk 3]
```diff
@@ -66,13 +74,16 @@ public void configureAfterEvaluate(final MinicondaExtension miniconda, File cond
...
                     getProject().delete(miniconda.getBootstrapDirectory());
+                    LOG.debug("Deleted BootstrapPython dir: {}", miniconda.getBootstrapDirectory());
                 }
             }
         });
         onlyIf(new Spec<Task>() {
             @Override
             public boolean isSatisfiedBy(Task task) {
-                return !miniconda.getBootstrapDirectory().exists();
+                boolean directoryExists = !miniconda.getBootstrapDirectory().exists();
```
Comment (punya) [inline src/main/java/com/palantir/python/miniconda/tasks/BootstrapPython.java]: Should this variable be called `directoryDoesNotExist`?

Comment (mnazbro) [inline src/main/java/com/palantir/python/miniconda/tasks/BootstrapPython.java]: Yes will change

**Comment by mnazbro** [comment]
@punya Ready for another CR

**Pull request merged**